src/Sylius/Bundle/CoreBundle/Controller/ProductController.php

```diff
@@ -196,6 +196,58 @@ public function historyAction(Request $request)
         return $this->handleView($view);
     }
 
+    public function stockAction(Request $request)
+    {
+        $product = $this->findOr404($request);
+
+        $form = $this->createForm('sylius_product_stock', null, array('product' => $product));
+
+        if ($form->handleRequest($request)->isValid()) {
+            $data = $form->getData();
+
+            $quantity = $data['quantity'];
+
+            if (!array_key_exists('variant', $data)) {
+                $variant = $product->getMasterVariant();
+            } else {
+                $variant = $data['variant'];
+            }
+
+            $stockItem = $this->get('sylius.repository.stock_item')
+                ->findByStockableAndLocation($variant, $data['location'])
+            ;
+
+            if (null === $stockItem) {
+                $stockItem = $this->get('sylius.factory.stock_item')
+                    ->create($variant, $data['location'])
+                ;
+            }
+
+            $operator = $this->get('sylius.inventory_operator');
+
+            if ($quantity > 0) {
+                $operator->increase($stockItem, $quantity);
```

Comment:
Fails with:

```
Catchable Fatal Error: Argument 1 passed to Sylius\Component\Inventory\Operator\InventoryOperator::increase() must implement interface Sylius\Component\Inventory\Model\StockableInterface, instance of Sylius\Component\Inventory\Model\StockItem given
```

Should we remove entire action at this point from this PR and focus on updating operator later? I feel like this PR is already large. :)